File: Solved/C++/CurseTheDarkness.cpp
#include <iostream>
#include <math.h>
using namespace std;

double square(double a){
	return a * a;
}

int main(void)
{
	int tc,n,canLight;
	double ux,uy,cx,cy;
	cin>>tc;

	while(tc--)
	{
		cin>>ux>>uy;
		cin>>n;
		canLight=0;
		for (int j = 0; j < n; ++j)
		{
			cin>>cx>>cy;

			double d=sqrt(square(ux-cx) + square(uy-cy));
			if (d<=8.0000000)
			{
				canLight++;
			}
		}

		if (canLight!=0)
		{
			cout<<"light a candle\n";
		}
		else
		{
			cout<<"curse the darkness\n";
		}
	}
	return 0;
}

File: Unsolved!/Python/CurseTheDarkness.py
## The whole program is right but it gives a NameError after 1 right testcase! Need help!

import math
tc = int(input())

def checkIfWithin(x,y,x1,y1):
    a1,b1 = (x,y),(x1,y1)
    dist = math.sqrt(sum([(a - b) ** 2 for a, b in zip(a1, b1)]))
    return dist


for i in range(tc):
    ux,uy = map(float, input().split())
    n = int(input())
    for j in range(n):
        candles=[]
        canLight=False
        cx, cy = map(float, input().split())
        candles.append(checkIfWithin(ux,uy,cx,cy))
        for x in candles:
            if x <=8.0:
                canLight=True
                break
    if canLight==True:
        print('light a candle')
    else:
        print('curse the darkness')
